server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/MailboxUpdateRequest.java

```diff
@@ -69,7 +61,11 @@ public Builder name(String name) throws MailboxException {
         }
 
         public Builder parentId(String parentId) {
-            this.parentId = Optional.ofNullable(parentId);
+            if (parentId == null) {
+                this.parentId = null;
```

Comment:
We have to check different kind of input values for parentId:
- the parentId is not given -> Optional.empty
- the parentId is null (set this mailbox as a root) -> null
- the parentId is given -> Optional.of